### PR TITLE
feature: Create Helm charts for docx/pdf-converter images. Touches: cksource/cs#12191.

### PR DESCRIPTION
**Pull request by sgreczyn-cksource**
The new Helm charts allow to run converters.

### CODE REVIEW

File: kubernetes/helm/ckeditor-docx-converter/.helmignore [hunk 1]
```diff
@@ -0,0 +1,23 @@
+# Patterns to ignore when building packages.
+# This supports shell glob matching, relative path matching, and
```
Comment (godai78) [inline kubernetes/helm/ckeditor-docx-converter/.helmignore:2]: global?

File: kubernetes/helm/ckeditor-docx-converter/README.md [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+>:warning: By default, the chart installs the CKEditor Word Services
+>on-premises with the "latest" tag. If you are using this chart for a production
+>environment, it's strongly recommended to change the container image tag to a
+>numeric representation of the version you want to install.
+```sh
+
+helm install ckeditor-docx-converter ./ckeditor-docx-converter \
+    --set image.tag="latest" \
+    --set server.secret.data.LICENSE_KEY="" \
+    --set server.ingress.hosts[0].host="test.example"
```
Comment (kbetanski-cksource) [inline kubernetes/helm/ckeditor-docx-converter/README.md:38]: Ingress is disabled by default, so there's no point in passing the hostname only. Also, there's no server object in yaml config

```suggestion
    --set ingress.hosts[0].host="test.example" \
    --set ingress.enabled=true
```

File: kubernetes/helm/ckeditor-pdf-converter/README.md [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+>:warning: By default, the chart installs the CKEditor PDF Converter
+>on-premises with the "latest" tag. If you are using this chart for a production
+>environment, it's strongly recommended to change the container image tag to a
+>numeric representation of the version you want to install.
+```sh
+
+helm install ckeditor-pdf-converter ./ckeditor-pdf-converter \
+    --set image.tag="latest" \
+    --set server.secret.data.LICENSE_KEY="" \
+    --set server.ingress.hosts[0].host="test.example"
```
Comment (kbetanski-cksource) [inline kubernetes/helm/ckeditor-pdf-converter/README.md:38]: Same case

```suggestion
    --set ingress.hosts[0].host="test.example" \
    --set ingress.enabled=true
```

File: kubernetes/helm/ckeditor-docx-converter/README.md [hunk 4]
```diff
@@ -0,0 +1,48 @@
+# CKEditor Word Services On-Premises Helm chart
+
+Use this Helm chart to provision CKEditor Word Services on your
+Kubernetes cluster.
+
+## Minimum requirements
+- 1 CPU Core
+- 512MB RAM
```
Comment (kbetanski-cksource) [inline kubernetes/helm/ckeditor-docx-converter/README.md:8]: The default setting provisions two instances of the converter

```suggestion
## Minimum requirements
- 2 CPU Cores
- 1024MB RAM
```

File: kubernetes/helm/ckeditor-pdf-converter/README.md [hunk 5]
```diff
@@ -0,0 +1,48 @@
+# CKEditor PDF Converter On-Premises Helm chart
+
+Use this Helm chart to provision CKEditor PDF Converter on your
+Kubernetes cluster.
+
+## Minimum requirements
+- 1 CPU Core
+- 512MB RAM
```
Comment (kbetanski-cksource) [inline kubernetes/helm/ckeditor-pdf-converter/README.md:8]: The default setting provisions two instances of the converter

```suggestion
## Minimum requirements
- 2 CPU Cores
- 1024MB RAM
```

File: kubernetes/helm/ckeditor-pdf-converter/README.md [hunk 6]
```diff
@@ -0,0 +1,48 @@
...
+- Kubernetes 1.19+
+- Helm v3
+
+The default configuration of running this service requires the reservation of 1
+CPU core and 512MB of RAM in the cluster. For more information about resources
+usage look here:
```
Comment (kbetanski-cksource) [inline kubernetes/helm/ckeditor-pdf-converter/README.md:14]:
```suggestion
The default configuration of running this service requires the reservation of 2
CPU cores and 1024MB of RAM in the cluster. For more information about resources
usage look here:
```

File: kubernetes/helm/ckeditor-docx-converter/README.md [hunk 7]
```diff
@@ -0,0 +1,48 @@
...
+- Kubernetes 1.19+
+- Helm v3
+
+The default configuration of running this service requires the reservation of 1
+CPU core and 512MB of RAM in the cluster. For more information about resources
+usage look here:
```
Comment (kbetanski-cksource) [inline kubernetes/helm/ckeditor-docx-converter/README.md:14]:
```suggestion
The default configuration of running this service requires the reservation of 2
CPU cores and 1024MB of RAM in the cluster. For more information about resources
usage look here:
```

**Review by kbetanski-cksource — CHANGES_REQUESTED** [review]

**Comment by kbetanski-cksource** [comment]
IMO you can remove tests and `NOTES.TXT` files

File: kubernetes/helm/ckeditor-docx-converter/values.yaml [hunk 8]
```diff
@@ -0,0 +1,85 @@
...
+  className: ""
+  annotations:
+    {}
+    # kubernetes.io/ingress.class: nginx
+    # kubernetes.io/tls-acme: "true"
+    # nginx.ingress.kubernetes.io/proxy-body-size: 10m
+    # nginx.ingress.kubernetes.io/proxy-read-timeout: "3600"
+    # nginx.ingress.kubernetes.io/proxy-send-timeout: "3600"
+  hosts:
+    - host: ckeditor-cs.organization.test
```
Comment (kbetanski-cksource) [inline kubernetes/helm/ckeditor-docx-converter/values.yaml:59]:
```suggestion
    - host: ckeditor-docx-converter.organization.test
```

File: kubernetes/helm/ckeditor-pdf-converter/values.yaml [hunk 9]
```diff
@@ -0,0 +1,85 @@
...
+  className: ""
+  annotations:
+    {}
+    # kubernetes.io/ingress.class: nginx
+    # kubernetes.io/tls-acme: "true"
+    # nginx.ingress.kubernetes.io/proxy-body-size: 10m
+    # nginx.ingress.kubernetes.io/proxy-read-timeout: "3600"
+    # nginx.ingress.kubernetes.io/proxy-send-timeout: "3600"
+  hosts:
+    - host: ckeditor-cs.organization.test
```
Comment (kbetanski-cksource) [inline kubernetes/helm/ckeditor-pdf-converter/values.yaml:59]:
```suggestion
    - host: ckeditor-pdf-converter.organization.test
```

**Review by kbetanski-cksource — CHANGES_REQUESTED** [review]

File: kubernetes/helm/ckeditor-pdf-converter/.helmignore [hunk 10]
```diff
@@ -0,0 +1,23 @@
+# Patterns to ignore when building packages.
+# This supports shell glob matching, relative path matching, and
```
Comment (godai78) [inline kubernetes/helm/ckeditor-pdf-converter/.helmignore:2]: Again, is "glob" legit?

**Review by godai78 — APPROVED** [review]
LGTM

File: kubernetes/helm/ckeditor-pdf-converter/README.md (hunk 6 above)
Comment (godai78) [reply kubernetes/helm/ckeditor-pdf-converter/README.md]: My review changed this part of the content, so these will need to be reintroduced by hand.

File: kubernetes/helm/ckeditor-docx-converter/.helmignore (hunk 1 above)
Comment (sgreczyn-cksource) [reply kubernetes/helm/ckeditor-docx-converter/.helmignore:2]: glob its fine, its a kind of patterns used commonly in file paths

File: kubernetes/helm/ckeditor-pdf-converter/.helmignore (hunk 10 above)
Comment (sgreczyn-cksource) [reply kubernetes/helm/ckeditor-pdf-converter/.helmignore:2]: https://github.com/cksource/ckeditor-cs-on-premises-infrastructure/pull/15#discussion_r1042179297

File: kubernetes/helm/ckeditor-pdf-converter/README.md (hunk 6 above)
Comment (godai78) [reply kubernetes/helm/ckeditor-pdf-converter/README.md]: Fixed in commit.

File: kubernetes/helm/ckeditor-docx-converter/README.md (hunk 7 above)
Comment (godai78) [reply kubernetes/helm/ckeditor-docx-converter/README.md]: Fixed in commit.

**Pull request merged**